app/src/main/java/com/example/myweatherapp/FavoriteAdapter.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.myweatherapp
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.myweatherapp.interfaces.IListItem
+import com.example.myweatherapp.interfaces.OnItemClickListener
+import com.example.myweatherapp.interfaces.OnSourceClickListener
+import com.example.myweatherapp.models.NewsModel
+import com.example.myweatherapp.models.SourceModel
+import com.example.myweatherapp.viewHolders.NewsHolder
+import com.example.myweatherapp.viewHolders.SourceHolder
+
+open class FavoriteAdapter(private val onItemClickListener: OnItemClickListener) : RecyclerView.Adapter<NewsHolder>() {
```

Comment:
Why did you decide to add a new adapter? I do see that we display the same cells on both News and Favorites tabs. I do believe we should use the same adapter and that's it.
Also `open` keyword is not required here cause `open` used for the inheritance